### PR TITLE
[cli] add Doxygen tags for history commands

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

File: src/cli/cli_history.cpp [hunk 1]
```diff
@@ -71,6 +71,54 @@ otError History::ParseArgs(Arg aArgs[], bool &aIsList, uint16_t &aNumEntries) co
...
+ * @par
+ * Displays the unicast IPv6 address history in table or list format.
+ * @par
+ * Each table or list entry provides:
+ * * Age: Time elapsed since the command was issued, and given in the format:
+ *        `hours`:`minutes`:`seconds`:`milliseconds`
+ * * Event: Possible values are `Added` or `Removed`.
+ * * Address/Prefix Length: Unicast address with its prefix length (in bits).
+ * * Origin: Possible value are `thread`, `slaac`, `dhcp6`, or `manual`.
+ * * Scope:
```
Comment (jrhodie) [inline src/cli/cli_history.cpp:115]: @abtink  For "Scope,"  what are the possible values and what does each value mean? For "Preferred," what does this flag indicate? Also, what does the "Valid" Flag indicate?

File: src/cli/cli_history.cpp [hunk 2]
```diff
@@ -191,6 +287,56 @@ template <> otError History::Process<Cmd("ipmaddr")>(Arg aArgs[])
...
+ * Displays the neighbor history in table or list format.
+ * @par
+ * Each table or list entry provides:
+ * * Age: Time elapsed since the command was issued, and given in the format:
+ *        `hours`:`minutes`:`seconds`:`milliseconds`
+ * * Type: `Child` or `Router`.
+ * * Event: Possible values are `Added`, `Removed`, or `Changed`.
+ * * Extended Address
+ * * RLOC16
+ * * Mode: MLE link mode.
```
Comment (jrhodie) [inline src/cli/cli_history.cpp:335]: @abtink What are the possible values for MLE link mode?

File: src/cli/cli_history.cpp [hunk 3]
```diff
@@ -342,6 +569,56 @@ template <> otError History::Process<Cmd("router")>(Arg aArgs[])
...
+ * @par
+ * Displays the network info history in table or list format.
+ * @par
+ * Each table or list entry provides:
+ * * Age: Time elapsed since the command was issued, and given in the format:
+ *        `hours`:`minutes`:`seconds`:`milliseconds`
+ * * Role: Device role. Possible values are `router`, `child`, `detached`, or `disabled`.
+ * * Mode:
+ * * RLOC16
+ * * Partition ID.
```
Comment (jrhodie) [inline src/cli/cli_history.cpp:619]: @abtink For Partition ID, what would a value of 0 indicate?

File: src/cli/cli_history.cpp [hunk 4]
```diff
@@ -342,6 +569,56 @@ template <> otError History::Process<Cmd("router")>(Arg aArgs[])
...
+ *   most-recent entries specified. If this option is not used, all stored
+ *   entries are shown in the output.
+ * @par
+ * Displays the network info history in table or list format.
+ * @par
+ * Each table or list entry provides:
+ * * Age: Time elapsed since the command was issued, and given in the format:
+ *        `hours`:`minutes`:`seconds`:`milliseconds`
+ * * Role: Device role. Possible values are `router`, `child`, `detached`, or `disabled`.
+ * * Mode:
```
Comment (jrhodie) [inline src/cli/cli_history.cpp:617]: @abtink For "Mode," what are the possible values?

File: src/cli/cli_history.cpp (hunk 1 above)
Comment (abtink) [reply src/cli/cli_history.cpp:115]: These are well-known properties of IPv6 address. No need to describe in the docs. So can simply say:
```
  * Scope: IPv6 address scope
  * Preferred and valid flags.
```

Just for your own information:
- Scope -> https://en.wikipedia.org/wiki/IPv6_address#Address_scopes
- Lifetime flags (valid and preferred) -> https://en.wikipedia.org/wiki/IPv6_address#Address_lifetime

File: src/cli/cli_history.cpp (hunk 2 above)
Comment (abtink) [reply src/cli/cli_history.cpp:335]: Same as `mode`:
https://github.com/openthread/openthread/blob/666a9bd076adfee873fafc5354e0b309cb4f6dda/src/cli/cli.cpp#L4139-L4156

File: src/cli/cli_history.cpp (hunk 4 above)
Comment (abtink) [reply src/cli/cli_history.cpp:617]: Same as previous one.

File: src/cli/cli_history.cpp [hunk 5]
```diff
@@ -191,6 +287,56 @@ template <> otError History::Process<Cmd("ipmaddr")>(Arg aArgs[])
     return error;
 }
 
+/**
+ * @cli history neighbor
+ * @code
+ * | Age                  | Type   | Event     | Extended Address | RLOC16 | Mode | Ave RSS |
```
Comment (Vyrastas) [inline src/cli/cli_history.cpp:298]: Missing the command here, I assume this one is for `history neighbor`

File: src/cli/cli_history.cpp [hunk 6]
```diff
@@ -134,6 +182,54 @@ template <> otError History::Process<Cmd("ipaddr")>(Arg aArgs[])
...
+ * @par
+ * Displays the multicast IPv6 address history in table or list format.
+ * @par
+ * Each table or list entry provides:
+ * * Age: Time elapsed since the command was issued, and given in the format:
+ *        `hours`:`minutes`:`seconds`:`milliseconds`
+ * * Event: Possible values are `Subscribed` or `Unsubscribed`.
+ * * Multicast Address
+ * * Origin: Possible values are `Thread` or `Manual`.
+ * @sa otHistoryTrackerEntryAgeToString
```
Comment (Vyrastas) [inline src/cli/cli_history.cpp:235]: Related... can you figure out why the content for this function doesn't output correctly on openthread.io?  

https://openthread.io/reference/group/api-history-tracker#othistorytrackerentryagetostring

Needs a fix here on GitHub

**Review by Vyrastas — CHANGES_REQUESTED** [review]

**Review by Vyrastas — APPROVED** [review]
Looks great 👍

**Review by abtink — APPROVED** [review]
LGTM. 👍

**Review by jwhui — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**